operator/standalone/update.go

```diff
@@ -0,0 +1,386 @@
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      PostgresqlPodName,
+			Namespace: u.instance.Status.HelmChart.DeploymentNamespace,
+		},
+	}
+	err := u.client.Get(ctx, types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}, pod)
+	if err == nil || isStatusConditionTrue(pod.Status.Conditions, conditions.TypeReady) {
+		return true
+	}
+	return false
+}
+
+// markInstanceAsProgressing marks an instance as progressing by updating the status conditions.
+func (u *UpdateStandalonePipeline) markInstanceAsProgressing(ctx context.Context) error {
+	meta.SetStatusCondition(
+		&u.instance.Status.Conditions,
+		conditions.Builder().
+			With(conditions.Progressing()).
+			WithGeneration(u.instance).
+			Build(),
+	)
+	meta.SetStatusCondition(
+		&u.instance.Status.Conditions,
+		conditions.Builder().
+			With(conditions.Ready(metav1.ConditionFalse)).
+			WithGeneration(u.instance).
+			Build(),
+	)
+	return u.client.Status().Update(ctx, u.instance)
+}
+
+// markInstanceAsReady marks an instance immediately as ready by updating the status conditions.
+func (u *UpdateStandalonePipeline) markInstanceAsReady(ctx context.Context) error {
+	meta.SetStatusCondition(
+		&u.instance.Status.Conditions,
+		conditions.Builder().
+			With(conditions.Ready(metav1.ConditionTrue)).
+			WithGeneration(u.instance).
+			Build(),
+	)
+	meta.RemoveStatusCondition(&u.instance.Status.Conditions, conditions.TypeProgressing)
+	return u.client.Status().Update(ctx, u.instance)
+}
+
+//  fetchConnectionSecret gets the connection secret for this instance.
+func (u *UpdateStandalonePipeline) fetchConnectionSecret(ctx context.Context) error {
+	connectionSecret := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{
+		Name:      u.instance.Spec.WriteConnectionSecretToRef.Name,
+		Namespace: u.instance.Namespace,
+	}}
+	err := u.client.Get(ctx, types.NamespacedName{Name: connectionSecret.Name, Namespace: connectionSecret.Namespace}, connectionSecret)
+	if err != nil {
+		return err
+	}
+	u.connectionSecret = connectionSecret
+	return nil
+}
+
+// isRootUserNotConsistentWithSecretData checks whether data in secret is consistent with EnableSuperUser value
+func (u *UpdateStandalonePipeline) isRootUserNotConsistentWithSecretData(_ context.Context) bool {
+	enabledUser := u.instance.Spec.Parameters.EnableSuperUser
+	rootUserExists := u.connectionSecret.Data["POSTGRESQL_POSTGRES_PASSWORD"] != nil
+	// Changed only when both variables have different values
+	if (enabledUser || rootUserExists) && !(enabledUser && rootUserExists) {
+		return true
+	}
+	return false
+}
+
+// isStatusConditionTrue checks whether pod condition type is active from the pod conditions array
+func isStatusConditionTrue(podConditions []corev1.PodCondition, conditionType corev1.PodConditionType) bool {
+	for _, condition := range podConditions {
+		if condition.Type == conditionType {
+			return condition.Status == corev1.ConditionTrue
+		}
+	}
+	return false
+}
```

Comment:
I think it's important to mention the "pod", because there's also `meta.IsStatusConditionTrue`, but it doesn't work with pod conditions.
```suggestion
// isPodStatusConditionTrue checks whether pod condition type is active from the pod conditions array
func isPodStatusConditionTrue(podConditions []corev1.PodCondition, conditionType corev1.PodConditionType) bool {
	for _, condition := range podConditions {
		if condition.Type == conditionType {
			return condition.Status == corev1.ConditionTrue
		}
	}
	return false
}
```

Reply:
I think we might not need this function at all if we manage the restart of the instance via annotations instead of deleting the pod in the first place.